src/Sarif.UnitTests/RandomSarifLogGenerator.cs

```diff
@@ -105,7 +106,8 @@ public static IList<Result> GenerateFakeResults(Random random, List<string> rule
                             {
                                 FileLocation = new FileLocation
                                 {
-                                    Uri = filePaths[random.Next(filePaths.Count)]
+                                    Uri = filePaths[fileIndex],
+                                    FileIndex = fileIndex
```

Comment:
>FileIndex = fileIndex [](start = 36, length = 21)

The results matcher now only persists files that are observed via results. I will fix this later, the results matcher should retain all files from current run. Implementation as it stands exists to ensure we flow property bags associated with baseline files, when appropriate. It's easier for me to just fix the gap then to explain much more. :)

[https://github.com/Microsoft/sarif-sdk/issues/1235](https://github.com/Microsoft/sarif-sdk/issues/1235)